app/src/features/Mappings/MappingNameDialog.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useEffect, useState } from "react";
+
+import {
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogProps,
+  DialogTitle,
+  DialogContent,
+  TextField,
+  DialogActions,
+  makeStyles,
+  Typography,
+} from "@material-ui/core";
+import { FetchBaseQueryError } from "@rtk-incubator/rtk-query/dist";
+import clsx from "clsx";
+import { head } from "lodash";
+import { useTranslation } from "react-i18next";
+import { useParams } from "react-router-dom";
+
+import Alert from "common/components/Alert";
+import { apiValidationErrorFromResponse } from "services/api/errors";
+import {
+  ResourceRequest,
+  useApiResourcesRetrieveQuery,
+  useApiResourcesUpdateMutation,
+} from "services/api/generated/api.generated";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    padding: theme.spacing(3),
+  },
+  button: {
+    margin: theme.spacing(2),
+    textTransform: "none",
+  },
+  previousButton: {
+    color: theme.palette.text.secondary,
+    "&:hover": {
+      backgroundColor: "inherit",
+      color: theme.palette.text.primary,
+    },
+  },
+}));
+
+type MappingNameDialogProps = {
+  onSubmit?: (name: string) => void;
+} & DialogProps;
+
+const MappingNameDialog = (props: MappingNameDialogProps): JSX.Element => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const [name, setName] = useState("");
+  const [alert, setAlert] = useState<string | undefined>(undefined);
+  const { mappingId } = useParams<{
+    mappingId?: string;
+  }>();
+  const {
+    data: mapping,
+    isLoading: isMappingLoading,
+  } = useApiResourcesRetrieveQuery(
+    {
+      id: mappingId ?? "",
+    },
+    { skip: !mappingId }
+  );
+  const [
+    updateMapping,
+    { isLoading: isUpdateLoading },
+  ] = useApiResourcesUpdateMutation({});
+
+  useEffect(() => {
+    setName(mapping?.label ?? "");
```

Comment:
why this is needed?

Reply:
Because on first mount mapping could be undefined, so when it loads it would not set the input value if it is not configures in useEffect